### PR TITLE
Pass ECONNREFUSED to session (closes #446)

### LINKED ISSUE #446: DestinationRequest should emit fatal error on ECONNREFUSED
Now fatal error is emitted only if requests to a destination emits an ENOTFOUND. But if requested server is down, an another error - ECONNREFUSED - will be emitted. So we should consider ECONNREFUSED a fatal error too.

### PR DESCRIPTION
**Pull request by AndreyBelym**
\cc @inikulin @churkin

### CODE REVIEW

File: test/server/proxy-test.js [hunk 1]
```diff
@@ -795,5 +796,33 @@ describe('Proxy', function () {
                 done();
             });
         });
+
+        it('Should pass ECONNREFUSED error to session (GH-446)', function (done) {
+            freeport(function (error, port) {
```
Comment (inikulin) [inline test/server/proxy-test.js]: Tip: you can use awesome [endpoint-utils](https://github.com/inikulin/endpoint-utils) package that has promise-based functionality, so you will not need ugly callbacks :smile:

Comment (AndreyBelym) [inline test/server/proxy-test.js]: thanks

**Comment by inikulin** [comment]
lgtm

**Comment by inikulin** [comment]
@testcafe-build-bot \retest

**Comment by inikulin** [comment]
@testcafe-build-bot \retest

**Comment by AndreyBelym** [comment]
Something has been broken, [see test logs](https://saucelabs.com/u/testcafebot):
Unsupported OS/browser/version/device combo: OS: 'Windows 10', Browser: 'iexplore', Version: '11.0.', Device: 'unspecified'

**Comment by inikulin** [comment]
\cc @churkin

**Comment by churkin** [comment]
lgtm

**Pull request merged**